### PR TITLE
feat(codesign) provide a zip archive to install the codesign application as a windows service

### PR DESCRIPTION
**Pull request by alachambre**
closes [BS-18522](https://bonitasoft.atlassian.net/browse/BS-18522)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,4 +23,24 @@ Provide a REST endpoint used by our CI to sign windows binaries with our EV Cert
 * Go to our [Artifactory](http://repositories.rd.lan/maven/releases/org/bonitasoft/releng/bonita-codesign-windows/) to retrieve the latest tag
 * Start the service using cmd : `java -jar path/to/bonita-codesign-windows-<version>.jar`
 * Make sure the service is started on 8080 port as the URL on the CI are configured on this port.
-* The endpoint should available at `http://codesign.rd.lan:8080/sign`, it accepts `POST` method with a parameter `exeFile`
\ No newline at end of file
+* The endpoint should available at `http://codesign.rd.lan:8080/sign`, it accepts `POST` method with a parameter `exeFile`
+
+## Install the codesign application as a service
```
Comment (tbouffard) [inline README.md]: ❓ why not provide a zip packaging whith the exe and the configuration?
as a end user I do not want to perform all this manual operations to have a working solution

File: build.gradle [hunk 2]
```diff
@@ -75,7 +109,10 @@ publishing {
 	}
 
 	publications {
-		bootJava(MavenPublication) { artifact bootJar }
+		bootJava(MavenPublication) {
+			artifact bootJar
```
Comment (tbouffard) [inline build.gradle]: ❓  Are we sure that we still want to publish the jar ? isn't the zip enough?

File: build.gradle [hunk 3]
```diff
@@ -35,8 +36,11 @@ repositories {
 	jcenter()
 	maven { url 'http://repositories.rd.lan/maven/all/' }
```
Comment (tbouffard) [inline build.gradle]: should be on top of the list

File: README.md [hunk 4]
```diff
@@ -20,29 +20,12 @@ Provide a REST endpoint used by our CI to sign windows binaries with our EV Cert
 
 ## Retrieve and start the latest version
 
-* Go to our [Artifactory](http://repositories.rd.lan/maven/releases/org/bonitasoft/releng/bonita-codesign-windows/) to retrieve the latest tag
+* Go to our [Artifactory](http://repositories.rd.lan/maven/releases/org/bonitasoft/releng/bonita-codesign-windows/) to retrieve the jar from the latest tag
```
Comment (tbouffard) [inline README.md]: ❓ why talking about our internal repository which is not available outside Bonitasoft?  The code is in a public repository and should not refer to internal Bonitasoft resources

File: README.md [hunk 5]
```diff
@@ -20,29 +20,12 @@ Provide a REST endpoint used by our CI to sign windows binaries with our EV Cert
...
-   \<arguments>-jar "bonita-codesign-windows-1.0.2.jar"\</arguments>  
-\</service>
-
-* Ensure that the codesign jar, the exe file and the xml file are in the same folder
-* From a command line interface, run the following command (It requires administrator rights): _bonitaCodesign.exe install_
-* Go to the service pannel in the task manager, you should see your service in the service list. If so, then the service is now installed, and will be launched each time the system restart.
-* Log files should be generated in the same folder
-* You can find more details about the usage of the service here: [winsw home](https://github.com/kohsuke/winsw)
\ No newline at end of file
+* Go to our [Artifactory](http://repositories.rd.lan/maven/releases/org/bonitasoft/releng/bonita-codesign-windows/) to retrieve the zip archive from the latest tag
```
Comment (tbouffard) [inline README.md]: ❓ same remark as for the jar

File: README.md [hunk 6]
```diff
@@ -20,29 +20,12 @@ Provide a REST endpoint used by our CI to sign windows binaries with our EV Cert
 
 ## Retrieve and start the latest version
 
-* Go to our [Artifactory](http://repositories.rd.lan/maven/releases/org/bonitasoft/releng/bonita-codesign-windows/) to retrieve the latest tag
```
Comment (tbouffard) [inline README.md]: ❓ I would drop all this section and keep only the section about the zip distribution archive

File: build.gradle [hunk 7]
```diff
@@ -47,6 +51,36 @@ dependencies {
 	testCompile 'org.jetbrains.spek:spek-junit-platform-engine:1.1.5'
 	testCompile 'org.junit.platform:junit-platform-runner:1.0.0'
 	testCompile "org.jetbrains.kotlin:kotlin-test:$kotlin_version"
+
+    mypackaging "com.sun.winsw:winsw:2.1.2:bin@exe"
+}
+
+task createXml(type: ConfigureBonitaCodesignXmlTask) {
```
Comment (tbouffard) [inline build.gradle]: ❓ is this possible to rename the task into something like `createServiceConfigurationFile` for clarity?

File: build.gradle [hunk 8]
```diff
@@ -84,4 +121,27 @@ processResources {
...
+
+	def xmlFile
+	def argument
+
+    @TaskAction
+    def addArgument() {
+		File file = new File(xmlFile)
+		file.createNewFile()
+
+		String xml = "<service>\n" +
```
Comment (tbouffard) [inline build.gradle]: ℹ️ can be simplified into the following for clarity
```
file.text= """<service>
  <id>bonita-codesign</id>
  <name>Bonita codesign</name>
  <description>This service runs the Bonita codesign windows application.</description>
  <executable>java</executable>
  <arguments>$argument</arguments>
</service>
"""
```

File: build.gradle (hunk 7 above)
Comment (alachambre) [reply build.gradle]: yes

File: build.gradle (hunk 2 above)
Comment (alachambre) [reply build.gradle]: no

File: build.gradle [hunk 9]
```diff
@@ -84,4 +118,24 @@ processResources {
...
+class ConfigureBonitaCodesignXmlTask extends DefaultTask {
+
+	def xmlFile
+	def argument
+
+    @TaskAction
+    def addArgument() {
+		File file = new File(xmlFile)
+		file.createNewFile()
+		file.text = "<service>\n" +
```
Comment (rbioteau) [inline build.gradle]: groovy multistrings are here to help

**Pull request merged**